app/controllers/api/v1/reset_password_controller.rb

```diff
@@ -23,6 +24,47 @@ def create
 
         render_json data: { token: } # TODO: enable reset email sending.
       end
+
+      # POST /api/v1/reset_password/reset.json
+      # Expects: { user: {:token, :new_password} }
+      # Returns: { data: Array[serializable objects] , errors: Array[String] }
+      # Does: Validates the token and reset the user password.
+
+      def reset
+        new_password = params[:user][:new_password]
+
+        return render_json status: :bad_request if new_password.blank?
+
+        # Invalidating token and changing the password.
+        # TODO: optimise this.
+        return render_json status: :internal_server_error unless @user.invalidate_reset_token
```

Comment:
Should we make the password reset happen even when unable to invalidate the reset token?
> Note: Doing that will give the user (any one with the token) under the validity period the ability to infinitely reset the password.